src/snowflake/snowpark/files.py

```diff
@@ -118,124 +184,283 @@ def open_new_result(cls, mode: str = "w") -> SnowflakeFile:
             from_result_api=True,
         )
 
+    def _raise_if_not_read(self) -> None:
+        """
+        Internal function to validate read mode of the file object before performing an IO operation.
+        """
+        if self._mode not in READ_MODES:
+            raise UnsupportedOperation(f"Not readable mode={self._mode}")
+
+    def _raise_if_not_write(self) -> None:
+        """
+        Internal function to validate write mode of the file object before performing a IO operation.
+        """
+        if self._mode not in WRITE_MODES:
+            raise UnsupportedOperation(f"Not writable mode={self._mode}")
```

Comment:
nit, you can improve the error experience by letting users know what are the supported mode in the err msg